app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -25,7 +25,8 @@ class Api::Rhsm::CandlepinProxiesController < Api::V2::ApiController
 
     before_action :check_content_type, :except => :async_hypervisors_update
 
-    skip_before_action :set_taxonomy, :only => [:rhsm_index, :consumer_activate, :consumer_create]
+    prepend_before_action :convert_owner_to_organization_id, :except =>  [:hypervisors_update, :async_hypervisors_update], :if => lambda { params.key?(:owner) }
+    prepend_before_action :convert_organization_label_to_id, :only => [:rhsm_index, :consumer_activate, :consumer_create], :if => lambda { params.key?(:organization_id) }
```

Comment:
`prepend_before_action` add callback to the beginning of the chain therefore even before `authorize` which triggers `set_taxnomy`, so by these two convert filers, we make sure we find the corrent organization_id using existing logic from `find_organization`

Reply:
however note that there's `set_organization_id` before_action that again overrides organization_id param if owner was specified so when we get to the action itself, it contains the org label again, this will be important later